### PR TITLE
Merging in minor misc bug fixes and the recipe reader

### PR DESCRIPTION
**Pull request by FlanFlanagan**
Last one!

### CODE REVIEW

File: cyclist/src/edu/utexas/cycic/OutPut.java [hunk 1]
```diff
@@ -4,7 +4,6 @@
 import java.util.ArrayList;
 import java.util.Base64;
 import java.util.Map;
-import java.util.Optional;
```
Comment (gonuke) [inline cyclist/src/edu/utexas/cycic/OutPut.java]: I think we need this line!

**Comment by gidden** [comment]
I have confirmed save/load now can resave

**Comment by gidden** [comment]
hi @FlanFlanagan, it looks like the "number" box in the institution pane still has an issue. I now see the full render of "Add" when I open the box, but now the "number" field is too small to actually show anything. It also looks like the field may not be updating when you type, only when you click the add button?

**Comment by gidden** [comment]
facility forms remember user level, nice!

**Comment by gidden** [comment]
Hi @FlanFlanagan, I don't think the null commodity treatment is quite right (though I'm not sure the best way to fix it). What I see now is that when I choose "New Commodity", an empty entry pops up in the "Simulation Commodities" pane, but the choice on the facility archetype does not update if I write anything there. Again, I'm not sure exactly what should happen instead..

**Comment by gidden** [comment]
Hi @FlanFlanagan, since we're getting up very close to the wire, would it be possible to split this PR into different PRs based on which issue is being addressed? That way we can selectively update features that definitely work. Let me know if you'd like any guidance -- I'm happy to help.

File: cyclist/src/edu/utexas/cycic/InstitutionView.java [hunk 2]
```diff
@@ -108,6 +109,7 @@ public void replaceSelection(String text) {
 				}
 			}
 		};
+		protoNumber.setMaxWidth(20);
```
Comment (gonuke) [inline cyclist/src/edu/utexas/cycic/InstitutionView.java]: I think this is too narrow.

File: cyclist/src/edu/utexas/cycic/InstitutionView.java [hunk 3]
```diff
@@ -49,7 +49,8 @@ public InstitutionView() {
 
 		ListView<String> facilityList = new ListView<String>();
         facilityList.setOrientation(Orientation.VERTICAL);
-        facilityList.setMinHeight(25);
+        facilityList.autosize();
+        facilityList.setMaxWidth(100);
```
Comment (gonuke) [inline cyclist/src/edu/utexas/cycic/InstitutionView.java]: Too narrow for many applications

**Comment by gonuke** [comment]
@gidden - I'm not sure how easy it will be to untanggle this - may be better to push through.

**Comment by gonuke** [comment]
What if we push this branch as a feature branch to cyclus/cyclist2 and then we can collaborate on it with direct pushes to the branch, and a new PR

**Comment by gonuke** [comment]
replacing this with #122 for collaborative development

**Comment by gonuke** [comment]
This is good now @FlanFlanagan - merging.

**Pull request merged**